stl/inc/source_location

```diff
@@ -25,7 +25,11 @@ _STD_BEGIN
 _EXPORT_STD struct source_location {
     _NODISCARD static consteval source_location current(const uint_least32_t _Line_ = __builtin_LINE(),
         const uint_least32_t _Column_ = __builtin_COLUMN(), const char* const _File_ = __builtin_FILE(),
+#if defined(__clang__) || defined(__EDG__) // TRANSITION, DevCom-10199227 and LLVM-58951
         const char* const _Function_ = __builtin_FUNCTION()) noexcept {
+#else // ^^^ workaround / no workaround vvv
+        const char* const _Function_ = __builtin_FUNCSIG()) noexcept {
+#endif // TRANSITION, DevCom-10199227 and LLVM-58951

```

Comment:
We should generally (in my understanding) be making sure that we don't put mismatched brackets in a `#if` block; instead, we should do something like:

```suggestion
#if defined(__clang__) || defined(__EDG__) // TRANSITION, DevCom-10199227 and LLVM-58951
        const char* const _Function_ = __builtin_FUNCTION()
#else // ^^^ workaround / no workaround vvv
        const char* const _Function_ = __builtin_FUNCSIG()
#endif // TRANSITION, DevCom-10199227 and LLVM-58951
        ) noexcept {
```